typecheck/typecheck.go

```diff
@@ -2728,29 +2727,32 @@ func (c *Checker) exprPartial(e expr.Expr, hint typeHint) (p partial) {
 			c.errorfmt("TODO index %T", lt)
 			return p
 		}
-		if atTyp := c.memory.Method(lt, "At"); atTyp != nil {
-			want := "At(i, j int) T"
-			if len(e.Indicies) == 1 {
-				want = "At(i int) T"
+
+		/*
+			if atTyp := c.memory.Method(lt, "At"); atTyp != nil {
+				want := "At(i, j int) T"
+				if len(e.Indicies) == 1 {
+					want = "At(i int) T"
+				}
+				if dim := len(atTyp.Params.Elems); dim == 0 || dim > 2 || dim != len(e.Indicies) ||
+					atTyp.Params.Elems[0] != tipe.Int || (dim == 2 && atTyp.Params.Elems[1] != tipe.Int) ||
+					len(atTyp.Results.Elems) != 1 {
+					p.mode = modeInvalid
+					c.errorfmt("cannot slice type %s, expecting method %s but type has %v", left.typ, want, atTyp)
+					return p
+				}
+				p.mode = modeVar
+				p.typ = atTyp.Results.Elems[0]
+				return p
 			}
-			if dim := len(atTyp.Params.Elems); dim == 0 || dim > 2 || dim != len(e.Indicies) ||
-				atTyp.Params.Elems[0] != tipe.Int || (dim == 2 && atTyp.Params.Elems[1] != tipe.Int) ||
-				len(atTyp.Results.Elems) != 1 {
+			if setTyp := c.memory.Method(lt, "Set"); setTyp != nil {
 				p.mode = modeInvalid
-				c.errorfmt("cannot slice type %s, expecting method %s but type has %s", left.typ, want, atTyp)
+				c.errorfmt("TODO Set index")
 				return p
 			}
-			p.mode = modeVar
-			p.typ = atTyp.Results.Elems[0]
-			return p
-		}
-		if setTyp := c.memory.Method(lt, "Set"); setTyp != nil {
-			p.mode = modeInvalid
-			c.errorfmt("TODO Set index")
-			return p
-		}
 
-		panic(fmt.Sprintf("typecheck.expr TODO Index: %s", format.Debug(e))) //, format.Debug(tipe.Underlying(left.typ))))
+			panic(fmt.Sprintf("typecheck.expr TODO Index: %s", format.Debug(e))) //, format.Debug(tipe.Underlying(left.typ))))
+		*/
```

Comment:
That's the largest chunk of unreachable code. All others are trivial. Probably that's not intended.